### PR TITLE
[v0.11.2] Sync release-0.11 branch with needed cherry-picks for TMC integration

### PR DESCRIPTION
**Pull request by navidshaikh**
[v0.11.2] Sync release-0.11 branch with needed cherry-picks for TMC integration

### CODE REVIEW

File: pkg/v1/tkg/client/validate_test.go [hunk 1]
```diff
@@ -1089,6 +1089,134 @@ var _ = Describe("Validate", func() {
...
+
+					coreDNSIP, _ := tkgConfigReaderWriter.Get(constants.ConfigVariableCoreDNSIP)
+					Expect(coreDNSIP).Should(Equal("::a"))
+		Context("Network Separation configuration and validation", func() {
+			It("should allow empty network separation configurations", func() {
+				validationError := tkgClient.ConfigureAndValidateManagementClusterConfiguration(initRegionOptions, true)
+				Expect(validationError).NotTo(HaveOccurred())
+			})
+
+			Context("Avi Management Cluster Service Engine Group", func() {
```
Comment (anujc25) [inline pkg/v1/tkg/client/validate_test.go:1181]: Looks like a bad merge/cherry-pick here as braces are not matching.

File: Makefile [hunk 2]
```diff
@@ -59,18 +59,18 @@ CUSTOM_NPM_REGISTRY ?= $(shell git config tkg.npmregistry)
 
 # TKG Compatibility Image repo and  path related configuration
 ifndef TKG_DEFAULT_IMAGE_REPOSITORY
-TKG_DEFAULT_IMAGE_REPOSITORY = "projects-stg.registry.vmware.com/tkg"
+TKG_DEFAULT_IMAGE_REPOSITORY = "projects.registry.vmware.com/tkg"
 endif
 ifndef TKG_DEFAULT_COMPATIBILITY_IMAGE_PATH
 # TODO change it to "tkg-compatibility" once the image is pushed to registry
-TKG_DEFAULT_COMPATIBILITY_IMAGE_PATH = "framework-zshippable/tkg-compatibility"
+TKG_DEFAULT_COMPATIBILITY_IMAGE_PATH = "tkg-compatibility"
```
Comment (navidshaikh) [inline Makefile:66]: gotta revert this and `pkg/v1/tkg/tkgconfigpaths/zz_bundled_default_bom_files_configdata.go` file change as those are downstream specific (being carried as patch downstream) - 

so revert https://github.com/vmware-tanzu/tanzu-framework/pull/1762/commits/9ebe78cf6f640f71ceae2dc49c71d6eba02b3571

**Comment by anujc25** [comment]
//usebom:tkg-compatibility:v5

File: Makefile (hunk 2 above)
Comment (vuil) [reply Makefile]: done

**Review by anujc25 — APPROVED** [review]

**Pull request merged**